src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestHandler.cs

```diff
@@ -228,31 +237,37 @@ public void Close()
         /// <summary>
         /// Update the test adapter paths provided through run settings to be used by the test plugin cache.
         /// </summary>
-        /// <param name="runSettings">
-        /// The run Settings.
+        /// <param name="payload">
+        /// The before test run start payload
         /// </param>
-        private void AddExtensionAssemblies(string runSettings)
+        private void AddExtensionAssemblies(BeforeTestRunStartPayload payload)
         {
             try
             {
-                IEnumerable<string> customTestAdaptersPaths = RunSettingsUtilities.GetTestAdaptersPaths(runSettings);
-                if (customTestAdaptersPaths != null)
+                var customTestAdaptersPaths = RunSettingsUtilities.GetTestAdaptersPaths(payload.SettingsXml);
+                var datacollectorSearchPaths = new List<string>();
+                foreach (var source in payload.Sources)
                 {
-                    var fileHelper = new FileHelper();
+                    datacollectorSearchPaths.Add(Path.GetDirectoryName(source));
+                }
+
+                datacollectorSearchPaths.AddRange(customTestAdaptersPaths);
 
+                if (datacollectorSearchPaths != null)
```

Comment:
This is redundant here, rather check for the customTestAdapterPaths above.

Reply:
Done